nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/IdentifyMimeType.java

```diff
@@ -233,53 +239,62 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
 
         final ComponentLog logger = getLogger();
-        final AtomicReference<String> mimeTypeRef = new AtomicReference<>(null);
-        final String filename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
-
-        session.read(flowFile, new InputStreamCallback() {
-            @Override
-            public void process(final InputStream stream) throws IOException {
-                try (final InputStream in = new BufferedInputStream(stream);
-                     final TikaInputStream tikaStream = TikaInputStream.get(in)) {
-                    Metadata metadata = new Metadata();
-
-                    if (filename != null && context.getProperty(USE_FILENAME_IN_DETECTION).asBoolean()) {
-                        metadata.add(TikaCoreProperties.RESOURCE_NAME_KEY, filename);
-                    }
-                    // Get mime type
-                    MediaType mediatype = detector.detect(tikaStream, metadata);
-                    mimeTypeRef.set(mediatype.toString());
-                }
+
+        final String mediaTypeString;
+        final String extension;
+        final Charset charset;
+
+        try (final InputStream flowFileStream = session.read(flowFile);
+             final TikaInputStream tikaStream = TikaInputStream.get(flowFileStream)) {
+            final String filename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
+
+            Metadata metadata = new Metadata();
+            if (filename != null && context.getProperty(USE_FILENAME_IN_DETECTION).asBoolean()) {
+                metadata.add(TikaCoreProperties.RESOURCE_NAME_KEY, filename);
             }
-        });
 
-        String mimeType = mimeTypeRef.get();
+            final MediaType mediaType = detector.detect(tikaStream, metadata);
+            mediaTypeString = mediaType.getBaseType().toString();
+            extension = lookupExtension(mediaTypeString, logger);
+            charset = identifyCharset(tikaStream, metadata, mediaType);
+        } catch (IOException e) {
+            throw new ProcessException("IOException thrown identifying mime-type of FlowFile content", e);
```

Comment:
It is not necessary to repeat `IOException` in the message since that will be part of the stack trace. Recommend the following:
```suggestion
            throw new ProcessException("Failed to identify MIME type from content stream", e);
```

Reply:
I adjusted the message as suggested.